pkg/menu/menu_test.go

```diff
@@ -0,0 +1,105 @@
+package menu
+
+import (
+	ui "github.com/gizak/termui/v3"
+	"strconv"
+	"testing"
+)
+
+func Test_newParagraph(t *testing.T) {
+	if err := ui.Init(); err != nil {
+		t.Fatal(err)
+	}
+	defer ui.Close()
+
+	testText := "newParagraph test"
+	p := newParagraph(testText, false, 0, 50, 3)
+	if testText != p.Text {
+		t.Errorf("incorrect value for p.Text. got: %v, want: %v", p.Text, testText)
+	}
+}
+
+func pressKey(ch chan ui.Event, input []string) {
+	var key ui.Event
+	for _, id := range input {
+		key = ui.Event{
+			Type: ui.KeyboardEvent,
+			ID:   id,
+		}
+		ch <- key
+	}
+}
+
+func Test_GetInput_Simple(t *testing.T) {
+	if err := ui.Init(); err != nil {
+		t.Fatal(err)
+	}
+	defer ui.Close()
+
+	input := newParagraph("", true, 0, 50, 3)
+	warning := newParagraph("", false, 3, 50, 3)
+	ui.Render(input)
+	ui.Render(warning)
+	testText := "test"
+
+	uiEvents := make(chan ui.Event)
+	go pressKey(uiEvents, []string{"t", "e", "s", "t", "<Enter>"})
+
+	isValid := func(input string) (string, bool) {
+		return "", true
+	}
+
+	inputText, err := processInput(input, warning, isValid, uiEvents)
+
+	if err != nil {
+		t.Errorf("Error: %v", err)
+	}
+	if inputText != testText {
+		t.Errorf("Incorrect value for input. got: %v, want: %v", inputText, testText)
+	}
+	if warning.Text != "" {
+		t.Errorf("Incorrect value for warning. got: %v, want nothing", warning.Text)
+	}
+}
+
+func Test_GetInput_Complicated(t *testing.T) {
```

Comment:
Nit: maybe rename to `TestGetInputInvalid` to indicate exactly what functionality youre testing?